test/visualTest.ts

```diff
@@ -583,22 +546,55 @@ describe("ChordChart", () => {
         const element: SVGElement = visualBuilder.slices[randomIndex];
         const svg: SVGElement = visualBuilder.svg;
         const datum: SelectableDataPoint = <SelectableDataPoint>select(element).datum();;
-        
+
         expect(datum.selected).toBeFalse();
         element.dispatchEvent(new MouseEvent("click"));
         expect(datum.selected).toBeTrue();
         svg.dispatchEvent(new MouseEvent("click"));
 
         for (let i = 0; i < visualBuilder.slices.length; i++) {
-          const dataPoint: SelectableDataPoint = <SelectableDataPoint>select(visualBuilder.slices[i]).datum();
-          const opacity: number = parseFloat(getComputedStyle(visualBuilder.slices[i]).opacity);
+          const { datum: dataPoint, opacity } = getDatumAndOpacity(visualBuilder.slices[i]);
           expect(dataPoint.selected).toBeFalse();
           expect(opacity).toBe(Behavior.FullOpacity);
         }
 
         done();
       });
     })
+
+    function testClickEvent(eventInitDict: MouseEventInit) {
+      const firstRandomIndex: number = Math.floor(Math.random() * visualBuilder.slices.length);
```

Comment:
random values are sometimes impossible to reproduce. Try not to use random in tests